pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -315,19 +315,43 @@ func (state PipelineRunState) getRetryableTasks(candidateTasks sets.String) []*R
 		if _, ok := candidateTasks[t.PipelineTask.Name]; ok {
 			var status *apis.Condition
 			var isCancelled bool
-			if t.TaskRun != nil {
+			switch {
+			case t.TaskRun != nil:
 				status = t.TaskRun.Status.GetCondition(apis.ConditionSucceeded)
 				isCancelled = t.TaskRun.IsCancelled()
 				if status != nil {
 					isCancelled = isCancelled || status.Reason == v1beta1.TaskRunReasonCancelled.String()
 				}
-
-			} else if t.Run != nil {
+			case len(t.TaskRuns) != 0:
+				isDone := true
+				isCancelled = false
+				for _, taskRun := range t.TaskRuns {
+					isDone = isDone && taskRun.IsDone()
+					c := taskRun.Status.GetCondition(apis.ConditionSucceeded)
+					if c.IsFalse() {
+						status = c
+						isCancelled = isCancelled || c.Reason == v1beta1.TaskRunReasonCancelled.String()
+					}
+				}
+				isCancelled = isCancelled && isDone
```

Comment:
I'm a bit confused about what's going on here. Does this mean a matrixed pipeline task is considered cancelled only if any taskrun is cancelled, and all of them are done? It looks like from the test cases we don't want to schedule any new matrixed tasks if any of them are cancelled. Could you update the docstring of this function to reflect how it works with matrix?

Reply:
@lbernick that's correct - if a pipelinetask has a matrix, iscancelled returns true if any run is cancelled and all other runs are done - this is also aligned with the iscancelled member function of pipelinetask:

https://github.com/tektoncd/pipeline/blob/8a3a8e5070fa99e157999e50c8fbd4711c786a5c/pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go#L237-L280

updated the docstring to reflect this